sos/plugins/crash_report.py

```diff
@@ -0,0 +1,256 @@
+                    line_contain_os = line
+                    self.crashed_kernel_name = line.split()[3]
+                    break
+            elif "crash" in line:
+                line_contain_crash = line
+                crash_found = True
+        if not crash_found or self.crashed_kernel_name == "":
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving crash from last cmd output:' +
+             '\n\n')
+            return -1
+        crashing_time = self.get_crash_time(line_contain_crash)
+        if not crashing_time:
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving time for last crash:' +
+             '[ ' + line_contain_crash + ' ]' + '\n\n')
+            return -1
+        ret = self.retrieve_path(self.dbg_vmlinux_path, "vmlinu",
+                                 self.crashed_kernel_name, crashing_time)
+        if ret:
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving debug linux:' +
+             '[ ' + self.crashed_kernel_name + ' ]' + ' inside path: ' +
+             '[ ' + self.dbg_vmlinux_path + ' ]' +
+             '\nPlease install the appropriate debug symbols to proceed ' +
+             'for your currently-running kernel, usually kernel-debuginfo' +
+             ' for RHEL and linux-image-$(uname -r)-dbgsym for Ubuntu\n\n')
+            return -1
+        ret = self.retrieve_path(self.vmcore_stnd_path, "vmcore", "",
+                                 crashing_time)
+        if ret:
+            self.message_to_log = (
+             '\n\n' + 'Failed in retrieving vmcore for kernel' +
+             '[ ' + self.crashed_kernel_name + ']' + ' inside path: ' +
+             '[ ' + self.vmcore_stnd_path + ' ]' + '\n\n')
+            return -1
+        return 0
+
+    # dump error log to output file
+    def dump_log_data(self):
+        try:
+            f = open(self.crashed_output_file, "a")
+        except IOError:
+            sys.stderr.write('\n\nFailed opening crash log file:[%s]\n\n'
+                             % self.crashed_output_file)
+            return -1
+        try:
+            f.write("\n\n[%s]\n\n" % self.message_to_log)
+        except:
+            sys.stderr.write('write to log file failed\n')
+            f.close()
+            return -1
+        f.close()
+        return 0
+
+    def postproc(self):
+        if os.path.exists(self.crashed_output_file):
+            os.remove(self.crashed_output_file)
+
+    def get_crash_report(self):
+        result = self.get_vmlinux_vmcore_path()
+        if result:
+            self.dump_log_data()
+            return
+
+        self.form_crash_script()
+        crash_full_command = ("crash" + " -s " + self.vmlinux_path + " " +
+                              self.vmcore_file_path)
+        self.message_to_log = 'crash command: ' + crash_full_command
+        self.dump_log_data()
+        print("\n\nCollecting crash report. This may take a while !!! ...\n\n")
```

Comment:
Hi Bryn,
Please suggest whether it is okay to keep this lines ?
Or should i remove this line ?
But keeping this line help user to wait so that user doesn't think that it is hanged or something .
Please suggest.
Will send patch once you respond.